[AIRFLOW-6562] Fix success/failed error in graph without dagrun

zhongjiajie opened the pull request:
Graph view now will raise an error when click
make task instance success/failed, with the
init state and no dag run, we should make some
hint instead of raise error in the webserver

---
Issue link: [AIRFLOW-6562](https://issues.apache.org/jira/browse/AIRFLOW-6562)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

zhongjiajie commented on airflow/www/views.py:
```diff
@@ -1258,15 +1258,12 @@ def _mark_task_instance_state(self, dag_id, task_id, origin, execution_date,
 
-        execution_date = timezone.parse(execution_date)
-
-        if not dag:
-            flash("Cannot find DAG: {}".format(dag_id))
```
Remove check dag and check task(in line 1267) due to when I add test to cover it I realize those two condition never reach with `task = dag.get_task(task_id)` in line 1258.

for dag check, I add test
```py
    def test_failed_invalid_dag(self):
        dag_id = "invalid"
        form = dict(
            task_id="run_this_last",
            dag_id=dag_id,
            execution_date=self.EXAMPLE_DAG_DEFAULT_DATE,
            upstream="false",
            downstream="false",
            future="false",
            past="false",
            origin="/graph?dag_id=example_bash_operator",
        )
        resp = self.client.post("failed", data=form, follow_redirects=True)
        self.check_content_in_response(f"Cannot find DAG: {dag_id}", resp)
```
And test failed due to `NoneType without function get_task`.

For task check, I add test
```py
    def test_success_invalid_task(self):
        task_id = "invalid"
        dag_id = "example_bash_operator"
        form = dict(
            task_id=task_id,
            dag_id=dag_id,
            execution_date=self.EXAMPLE_DAG_DEFAULT_DATE,
            upstream="false",
            downstream="false",
            future="false",
            past="false",
            origin="/graph?dag_id=example_bash_operator",
        )
        resp = self.client.post("success", data=form, follow_redirects=True)
        self.check_content_in_response(f"Cannot find task {task_id} in DAG {dag_id}", resp)
```
The test also failed due to `airflow.exceptions.TaskNotFound: Task invalid not found`

So I decide to remove them.

zhongjiajie commented on tests/www/test_views.py:
```diff
@@ -1083,7 +1149,7 @@ def test_with_execution_date_parameter_only(self):
         self.assert_run_is_in_dropdown_not_selected(self.runs[3], data)
 
-    def test_with_base_date_and_num_runs_parmeters_only(self):
+    def test_with_base_date_and_num_runs_parameters_only(self):
```
Typo here, Fix it

zhongjiajie commented on tests/www/test_views.py:
```diff
@@ -999,7 +1065,7 @@ def __init__(self, test, endpoint):
         self.endpoint = endpoint
 
-    def setUp(self):  # pylint: disable=invalid-name
+    def setup(self):
```
Both `setUp` and `tearDown` are function of `ViewWithDateTimeAndNumRunsAndDagRunsFormTester`, which a helper for more convenience in test, not inherit from `unittest.TestCase`. So I think is better to use lowercase and remove `pylint: disable`

ashb replied:
`setUp` is a special name in unittest - doesn't changing this mean it is no longer called automatically in every test?

ashb replied:
(oh I see, I'll need to look at this in a bigger viewer to see what's going on)

zhongjiajie reviewed (commented):
Add some hint for reviewer

zhongjiajie committed:
[AIRFLOW-7053] Fix success/failed error in graph without dagrun

Graph view now will raise an error when click
make task instance success/failed, with the
init state and no dag run, we should make some
hint instead of raise error in the webserver
